### PR TITLE
[FEAT] Add rejoining a finished game to view results

### LINKED ISSUE #274: [FEAT] Add rejoining a finished game to view results
Adding feature which allows players to rejoin their finished game using the game code to view results of the game.

The places most likely to change is the joining room part in the frontend and backend for searching the database for the finished lobby.

### LINKED ISSUE #300: [FEAT] Extract Google API key out of frontend
Extract the Google API key out of `index.html` in the frontend. This is because it is exposed to all users - it is hardcoded and poses a security threat.

To complete this, extract the key out and put it in the `.env` file for the backend to use.
Ensure that all usages of the google api key uses the environment variable, and nothing else breaks as a result.

### PR DESCRIPTION
**Pull request by jteo97**
--
name: Join to results page
about: If game code of finished game is entered, redirect to results page
title: [FEAT] Add rejoining a finished game to view results
labels: Frontend, Backend, Websockets
assignees: @jteo97 @Jmcl168 

---

Closes #274

### CODE REVIEW

File: backend/src/sockets/joinRoom.js [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
  */
 export function joinRoom(socket, io) {
-  socket.on('join_room', ({sessionId, name}) => {
+  socket.on('join_room', async ({sessionId, name}) => {
+
+    const finishedSession = await findSession(sessionId)
+    
+    // console.log(finishedSession.isFinished);
+    if (finishedSession && finishedSession.isFinished) {
+      io.to(socket.id).emit('skip-to-results', true)
```
Comment (parfei) [inline backend/src/sockets/joinRoom.js]: They typically name their event tags with underscores

File: backend/src/sockets/joinRoom.js [hunk 2]
```diff
@@ -27,5 +38,16 @@ export function joinRoom(socket, io) {
...
       io.to(sessionId).emit('update_restaurants', game.session.restaurants);
     }
+  }
   });
 }
+
+const findSession = async(sessionId) => {
+  try {
+    const finishedSession = await Session.findOne({truncCode: sessionId});
+    console.log(finishedSession.isFinished);
```
Comment (UltimaKuma) [inline backend/src/sockets/joinRoom.js]: Remove logs

File: backend/src/sockets/joinRoom.js [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
 export function joinRoom(socket, io) {
-  socket.on('join_room', ({sessionId, name}) => {
+  socket.on('join_room', async ({sessionId, name}) => {
+
+    const finishedSession = await findSession(sessionId)
+    
+    // console.log(finishedSession.isFinished);
+    if (finishedSession && finishedSession.isFinished) {
+      io.to(socket.id).emit('skip-to-results', true)
+      console.log("here");
```
Comment (UltimaKuma) [inline backend/src/sockets/joinRoom.js]: Remove this please

File: backend/src/sockets/joinRoom.js [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
  * @param {*} socket
  * @param {*} io
  */
 export function joinRoom(socket, io) {
-  socket.on('join_room', ({sessionId, name}) => {
+  socket.on('join_room', async ({sessionId, name}) => {
+
+    const finishedSession = await findSession(sessionId)
+    
+    // console.log(finishedSession.isFinished);
```
Comment (UltimaKuma) [inline backend/src/sockets/joinRoom.js]: Remove :)

**Review by UltimaKuma — APPROVED** [review]
Ran the application flow a few times and LGTM. Just a few logs and comments to remove.

File: backend/src/sockets/joinRoom.js [hunk 5]
```diff
@@ -1,11 +1,22 @@
...
+    if (finishedSession && finishedSession.isFinished) {
+      io.to(socket.id).emit('skip-to-results', true)
+      console.log("here");
+    } else {
+
```
Comment (parfei) [inline backend/src/sockets/joinRoom.js]: Do we want to abstract the else functionality out? :D I feel this function is quite large

**Review by parfei — APPROVED** [review]
Nice work :) Just some nitpicky comments that I have, otherwise all good

File: backend/src/sockets/joinRoom.js (hunk 3 above)
Comment (jteo97) [reply backend/src/sockets/joinRoom.js]: done

File: backend/src/sockets/joinRoom.js (hunk 2 above)
Comment (jteo97) [reply backend/src/sockets/joinRoom.js]: done

File: backend/src/sockets/joinRoom.js (hunk 1 above)
Comment (jteo97) [reply backend/src/sockets/joinRoom.js]: fixed

**Review by UltimaKuma — APPROVED** [review]
LGTM!

**Pull request merged**